test/evp_test.c

```diff
@@ -658,7 +663,17 @@ static int cipher_test_enc(EVP_TEST *t, int enc,
     EVP_CIPHER_CTX_set_padding(ctx, 0);
     t->err = "CIPHERUPDATE_ERROR";
     tmplen = 0;
-    if (!frag) {
+    if (expected->bit_len != 0) {
+        EVP_CIPHER_CTX_set_flags(ctx, EVP_CIPH_FLAG_LENGTH_BITS);
+        if (!EVP_CipherUpdate(ctx, tmp + out_misalign, &tmplen, in,
+                              expected->bit_len))
+            goto err;
+        if (tmplen != expected->bit_len)
+            goto err;
+        tmplen = (tmplen + 7) / 8;
+        if (expected->bit_len & 7)
+            tmp[out_misalign + tmplen - 1] &= ~0x7F >> (expected->bit_len & 7);
```

Comment:
I think the mask is not right, leaves one bit too much in the last byte,
consider `bit_len == 1` then this does `last_byte &= 0x7F >> 1 `